sheets/README.md

```diff
@@ -0,0 +1,104 @@
+# xPro Google Sheets
+
+**SECTIONS**
+1. [What Is This?](#what-is-this)
+1. [How It Works](#how-it-works)
+1. [Types of Sheets](#types-of-sheets)
+1. [Sheets Management](#sheets-management)
+1. [Development Setup](#development-setup)
+
+## What Is This?
+
+We've had a few customer service-related use cases that we have needed to support. One option in terms of implementation
+was to create several custom web UIs in xPro and give user access to all who needed it (CS representatives, finance 
+team members, etc.) This had the potential to become a persistent headache for developers, so we decided to support 
+these use cases through UIs that are very familiar to most users regardless of technical expertise: Google Sheets and
+Google Forms. Using various Google Drive and Sheets APIs, the following is now possible: 
+
+1. CS/finance users can make various requests (e.g.: enrollment code creation/assignment, enrollment deferrals) by filling 
+  out a Google Form or entering data into Google Sheets.
+1. Those requests are automatically processed by our app.
+1. Users can monitor the status of their requests directly in the spreadsheets.  
+
+## How It Works
+
+The basic workflow for most flavors of xPro sheets is roughly as follows:
+
+1. A user fills out a Google Form. That form submission is a request for some work to 
+be done, and it is automatically added to a form responses worksheet within a relevant spreadsheet.
+    - That submission also appears in the main worksheet within the spreadsheet automatically via a query.
+1. In response to the change in the spreadsheet, Google sends a file watch request to an endpoint in our app, which 
+  indicates to us that the spreadsheet has been changed, so we may have some new requests to handle.
+1. Our app scans the main worksheet of the spreadsheet for new or updated rows and takes the appropriate action if the
+  data is all valid (e.g.: issuing a refund, creating new enrollment codes, etc.)
+1. If a request was completed for some row, or if an error occurred, our app updates certain columns in the 
+  main worksheet to indicate the status of that request.  
+
+## Types of Sheets
+
+Right now there are three sheet types:
+
+1. Enrollment Code Request sheet (a.k.a. Coupon Request sheet)
+1. Enrollment Code Assignment sheets (which are created based on Enrollment Code Request sheet entries)
+1. Change of Enrollment Request sheet. This includes worksheets for the following:
+    1. Refund Requests
+    1. Deferral Requests
```

Comment:
Do you think it's OK to link to the sheets here? They do require authentication.